graphbolt/src/csc_sampling_graph.cc

```diff
@@ -221,17 +223,51 @@ torch::Tensor Pick(
       picked_neighbors =
           torch::randint(offset, offset + num_neighbors, {fanout}, options);
     } else {
-      picked_neighbors = torch::randperm(num_neighbors, options) + offset;
-      picked_neighbors = picked_neighbors.slice(0, 0, fanout);
+      picked_neighbors = torch::randperm(num_neighbors, options);
+      picked_neighbors = picked_neighbors.slice(0, 0, fanout) + offset;
     }
   }
   return picked_neighbors;
 }
 
+torch::Tensor NonUniformPick(
+    int64_t offset, int64_t num_neighbors, int64_t fanout, bool replace,
+    const torch::TensorOptions& options,
+    const torch::optional<torch::Tensor>& probs) {
+  torch::Tensor picked_neighbors;
+  auto local_probs = probs.value().slice(0, offset, offset + num_neighbors);
+  auto positive_probs_mask = (local_probs > 0);
+  if ((fanout == -1) || (num_neighbors <= fanout && !replace)) {
```

Comment:
Is this condition correct? Should it be `num_neighbors <= positive_probs_mask.size()`?

Reply:
right, let me fix